### DISCUSSION TITLE
Would it be possible to get the amount of ACPI battery voltage with Cosmos?

### DISCUSSION THREAD

**Discussion by elkaamee326** [CosmosOS/Cosmos#1800] [Q&A]
How would I be able to get the battery voltage of an ACPI Battery in Cosmos. Also, if this is not implemented yet how would this be done? Are there any resources online that are recommended (Ie: Registers, documentation, setup, etc...)?

**Comment by MishaTy** [comment]
Getting the battery voltage is very hard to implement in ACPI driver. First, you need to parse and execute the AML code in the ACPI tabels. Plus, Cosmos can only read a few parts of ACPI that is required for ACPI shutdown. You can read [this website](https://wiki.osdev.org/ACPI) about ACPI.

**Reply by elkaamee326** [reply]
Would this be possible to do so without having to write an AML interpreter? Would there be a way that I can just send a register or a command to the ACPI interface to return the voltage?

**Reply by MishaTy** [reply]
Nope. There is no standard command/register to get the battery voltage.

**Reply by elkaamee326** [reply]
Ok, thank you!

**Reply by ghost** [reply]
Excecute inline asm, dllimport c or cpp code? Not sure if these work but it's good to try.

**Reply by MishaTy** [reply]
ACPI bytecode is custom and you can't call it, you need an AML interpreter. By the way, dllimport is not supported
